File: src/main/java/cf/nirvandil/clientvds/util/SocketFactoryWithTimeout.java
package cf.nirvandil.clientvds.util;

import com.jcraft.jsch.SocketFactory;
import lombok.SneakyThrows;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.InetSocketAddress;
import java.net.Socket;

public class SocketFactoryWithTimeout implements SocketFactory {

    @SneakyThrows
    public Socket createSocket(String host, int port) {
        Socket socket = new Socket();
        int timeout = 5_000;
        socket.connect(new InetSocketAddress(host, port), timeout);
        return socket;
    }

    public InputStream getInputStream(Socket socket) throws IOException {
        return socket.getInputStream();
    }

    public OutputStream getOutputStream(Socket socket) throws IOException {
        return socket.getOutputStream();
    }
}

File: src/main/kotlin/cf/nirvandil/clientvds/util/SocketFactoryWithTimeout.kt
package cf.nirvandil.clientvds.util

import com.jcraft.jsch.SocketFactory
import lombok.SneakyThrows

import java.io.IOException
import java.io.InputStream
import java.io.OutputStream
import java.net.InetSocketAddress
import java.net.Socket

class SocketFactoryWithTimeout : SocketFactory {

    @SneakyThrows
    override fun createSocket(host: String, port: Int): Socket {
        return Socket().apply { connect(InetSocketAddress(host, port), TIMEOUT) }
    }

    @Throws(IOException::class)
    override fun getInputStream(socket: Socket): InputStream {
        return socket.getInputStream()
    }

    @Throws(IOException::class)
    override fun getOutputStream(socket: Socket): OutputStream {
        return socket.getOutputStream()
    }
}
